### PR TITLE
fix: Ledger disconnect polling

### PR DESCRIPTION
**Pull request by cvarley100**
# Description of change

Changes Ledger connection polling logic:
- Only polls after detecting a disconnect

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Still testing

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/shared/lib/ledger.ts [hunk 1]
```diff
@@ -6,62 +6,73 @@ import { api } from 'shared/lib/wallet'
...
+        stopPollingLedgerStatus()
+        onCancel()
+    }
+    const pollLedgerStatus = () => api.getLedgerDeviceStatus(ledgerSimulator, {
         onSuccess(response) {
             let _isLedgerConnected = response.payload?.type === LedgerStatus.Connected
             if (_isLedgerConnected) {
-                closePopup()
+                stopPollingLedgerStatus()
+                if (get(popupState).active) {
```
Comment (ingokpp) [inline packages/shared/lib/ledger.ts:28]: Opt: You could change this into the following:
`get(popupState).active && closePopup()`

File: packages/shared/routes/setup/ledger/views/Balance.svelte [hunk 2]
```diff
@@ -45,10 +45,8 @@
     let isCheckingForBalance = false
 
     function handleContinueClick() {
-        if (!legacyLedger) {
-            stopPollLedgerStatus()
-        } else {
-            stopPollLedgerLegacyStatus()
+        if (legacyLedger) {
+            stopPollingLedgerLegacyStatus()
```
Comment (ingokpp) [inline packages/shared/routes/setup/ledger/views/Balance.svelte:49]: Optional improvement :)
`legacyLedger && stopPollingLedgerLegacyStatus()`

File: packages/shared/components/popups/LedgerNotConnected.svelte [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
     import { closePopup } from 'shared/lib/popup'
+    import { stopPollingLedgerStatus } from 'shared/lib/ledger'
 
     export let message
     export let handleClose
     export let locale
 
     function handleCancelClick() {
+        stopPollingLedgerStatus()
         if ('function' === typeof handleClose) {
```
Comment (ingokpp) [inline packages/shared/components/popups/LedgerNotConnected.svelte:12]: You could also do this like this :) 
`typeof handleClose === 'function' && handleClose()`

File: packages/shared/lib/ledger.ts [hunk 4]
```diff
@@ -6,62 +6,91 @@ import { api } from 'shared/lib/wallet'
...
+            openLedgerNotConnectedPopup(false, onCancel)
+        }
+    }
+    getLedgerDeviceStatus(_onConnected, _onDisconnected, _onCancel)
+}
+
+export function pollLedgerDeviceStatus(pollInterval, _onConnected = () => { }, _onDisconnected = () => { }, _onCancel = () => { }) {
+    if (!polling) {
+        intervalTimer = setInterval(async () => {
+            getLedgerDeviceStatus(_onConnected, _onDisconnected, _onCancel)
```
Comment (begonaalvarezd) [inline packages/shared/lib/ledger.ts:62]: If we also call `getLedgerDeviceStatus(_onConnected, _onDisconnected, _onCancel)` right before starting the internal we can get the ledger status right at subscription time

File: packages/shared/routes/dashboard/wallet/views/Security.svelte [hunk 5]
```diff
@@ -92,22 +67,10 @@
...
-                }, 500)
-            },
-            onError(e) {
-                ledgerDeviceStatus = LedgerStatus.Disconnected
-                setTimeout(function () {
-                    isCheckingLedger = false
-                }, 500)
-            },
-        })
+        const _onComplete = () => (isCheckingLedger = false)
```
Comment (begonaalvarezd) [inline packages/shared/routes/dashboard/wallet/views/Security.svelte:72]: This is just a tinytiny design thing, perhaps we could delay setting this to `false` to allow the refresh icon to spin? 
```javascript
const _onComplete = () => {
    clearTimeout(spinnerTimeout)
    spinnerTimeout = setTimeout(() => (isCheckingLedger = false), 500)
}
```

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**